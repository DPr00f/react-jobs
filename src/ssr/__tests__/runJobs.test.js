import React from 'react'
import { mount } from 'enzyme'
import { withJob, runJobs } from '../'
import { Foo, resolveAfter, rejectAfter } from '../../../tools/tests/helpers'

const workTime = 10 // ms

describe('runJobs()', () => {
  it('should return a Promise', () => {
    const actual = runJobs(<div>foo</div>)
    expect(actual.then).toBeTruthy()
  })

  it('should render a job that succeeds', () => {
    const FooWithJob = withJob(() => resolveAfter(workTime, 'Hello world!'))(
      Foo,
    )
    const app = <FooWithJob />
    return runJobs(app).then(({ appWithJobs }) => {
      expect(mount(appWithJobs)).toMatchSnapshot()
    })
  })

  it('should render a job that fails', () => {
    const FooWithJob = withJob(() => rejectAfter(workTime, 'Poop!'))(Foo)
    const app = <FooWithJob />
    return runJobs(app).then(({ appWithJobs }) => {
      expect(mount(appWithJobs)).toMatchSnapshot()
    })
  })

  it('should not render a job that is deferred', () => {
    const FooWithJob = withJob(() => resolveAfter(workTime, 'Hello world!'), {
      defer: true,
    })(Foo)
    const app = <FooWithJob />
    return runJobs(app).then(({ appWithJobs }) => {
      expect(mount(appWithJobs)).toMatchSnapshot()
    })
  })
})
