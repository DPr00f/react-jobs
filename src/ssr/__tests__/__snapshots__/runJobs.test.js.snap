// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`runJobs() should not render a job that is deferred 1`] = `
<ServerProvider
  runJobsExecContext={
    Object {
      "getState": [Function],
      "registerJobState": [Function],
    }
  }
>
  <ClientProvider>
    <FooWithJobWithJobID>
      <FooWithJob
        jobInitState={
          Object {
            "completed": false,
            "inProgress": true,
          }
        }
        onJobProcessed={[Function]}
      >
        <Foo
          job={
            Object {
              "completed": false,
              "error": undefined,
              "inProgress": true,
              "result": undefined,
            }
          }
        >
          <div>
            In progress...
          </div>
        </Foo>
      </FooWithJob>
    </FooWithJobWithJobID>
  </ClientProvider>
</ServerProvider>
`;

exports[`runJobs() should render a job that fails 1`] = `
<ServerProvider
  runJobsExecContext={
    Object {
      "getState": [Function],
      "registerJobState": [Function],
    }
  }
>
  <ClientProvider>
    <FooWithJobWithJobID>
      <FooWithJob
        jobInitState={
          Object {
            "completed": true,
            "error": "Poop!",
            "inProgress": false,
            "result": undefined,
          }
        }
        onJobProcessed={[Function]}
      >
        <Foo
          job={
            Object {
              "completed": true,
              "error": "Poop!",
              "inProgress": false,
              "result": undefined,
            }
          }
        >
          <div>
            Error: 
            Poop!
          </div>
        </Foo>
      </FooWithJob>
    </FooWithJobWithJobID>
  </ClientProvider>
</ServerProvider>
`;

exports[`runJobs() should render a job that succeeds 1`] = `
<ServerProvider
  runJobsExecContext={
    Object {
      "getState": [Function],
      "registerJobState": [Function],
    }
  }
>
  <ClientProvider>
    <FooWithJobWithJobID>
      <FooWithJob
        jobInitState={
          Object {
            "completed": true,
            "error": undefined,
            "inProgress": false,
            "result": "Hello world!",
          }
        }
        onJobProcessed={[Function]}
      >
        <Foo
          job={
            Object {
              "completed": true,
              "error": undefined,
              "inProgress": false,
              "result": "Hello world!",
            }
          }
        >
          <div>
            Hello world!
          </div>
        </Foo>
      </FooWithJob>
    </FooWithJobWithJobID>
  </ClientProvider>
</ServerProvider>
`;
